### PR TITLE
chore: JSModule Action selector CE changes

### PR DESCRIPTION
**Pull request by rishabhrathod01**
## Description

- Action Selector CE changes

#### PR fixes following issue(s)
Fixes #

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Introduced new data retrieval functionality for JS module instances.

- **Enhancements**
  - Improved action creator helper functions to handle JS module instances.
  - Updated icon naming for better clarity and consistency.

- **Refactor**
  - Renamed functions to align with the expanded scope of handling both API queries and JS module actions.
  
- **Bug Fixes**
  - Adjusted logic in the Action Selector to correctly iterate over JS collections and module instances.

### CODE REVIEW

File: app/client/src/components/editorComponents/ActionCreator/helpers.tsx [hunk 1]
```diff
@@ -390,14 +391,21 @@ export function useModalDropdownList(handleClose: () => void) {
...
-export const getQueryInstanceIcon = (type: MODULE_TYPE): ReactNode => {
+export const getModuleInstanceIcon = (type: MODULE_TYPE): ReactNode => {
   if (type === MODULE_TYPE.QUERY) {
     return (
       <EntityIcon>
         <Icon name="module" />
       </EntityIcon>
     );
   }
+  if (type === MODULE_TYPE.JS) {
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/ActionCreator/helpers.tsx:402]: just add or statement to above block

File: app/client/src/components/editorComponents/ActionCreator/helpers.tsx [hunk 2]
```diff
@@ -568,6 +578,52 @@ export function getJSOptions(
         }
       }
     });
+
+    jsModuleInstances.forEach((jsModuleInstance) => {
+      if (!jsModuleInstance) return;
+      if (
+        jsModuleInstance.config.actions &&
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/ActionCreator/helpers.tsx:585]: you are going to add actions to jsModuleInstance in selector right?

File: app/client/src/ce/selectors/entitiesSelector.ts [hunk 3]
```diff
@@ -1424,6 +1424,14 @@ export const getQueryModuleInstances = () => {
   return [];
 };
 
+export const getJSModuleInstancesData = (_: AppState) => {
+  return [] as Array<{
```
Comment (ApekshaBhosale) [inline app/client/src/ce/selectors/entitiesSelector.ts:1428]: wer have added separate type for it. extend that or add this there. module instance constants i think

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/components/editorComponents/ActionCreator/helpers.tsx (hunk 2 above)
Comment (rishabhrathod01) [reply app/client/src/components/editorComponents/ActionCreator/helpers.tsx:585]: Yes.

File: app/client/src/components/editorComponents/ActionCreator/helpers.tsx (hunk 1 above)
Comment (rishabhrathod01) [reply app/client/src/components/editorComponents/ActionCreator/helpers.tsx:402]: Make sense.

File: app/client/src/ce/selectors/entitiesSelector.ts (hunk 3 above)
Comment (rishabhrathod01) [reply app/client/src/ce/selectors/entitiesSelector.ts:1428]: This is a particular type that no other selector would use and the return type of the selector itself is used in every place, so having the type defined for this in `module constant` doesn’t seem correct choice.

**Review by ApekshaBhosale — APPROVED** [review]

**Pull request merged**